packages/time-picker/src/TimePicker/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react"
+import {
+  useOverlay,
+  AriaOverlayProps,
+  DismissButton,
+} from "@react-aria/overlays"
+import { FocusScope } from "@react-aria/focus"
+import styles from "./Popover.module.scss"
+
+export const Popover: React.FunctionComponent<
+  AriaOverlayProps & { children: React.ReactNode }
+> = ({ isOpen, onClose, children, shouldCloseOnInteractOutside }) => {
+  const ref = React.useRef(null)
+  const { overlayProps } = useOverlay(
+    {
+      isOpen,
+      onClose,
+      shouldCloseOnBlur: true,
+      isDismissable: true,
+      shouldCloseOnInteractOutside,
+    },
+    ref
+  )
+  // Add a hidden <DismissButton> component at the end of the popover
+  // to allow screen reader users to dismiss the popup easily.
+  if (!isOpen) {
```

Comment:
If the hook takes in `isOpen`, does it not mean they have handled this for us? If not, would it make more sense (or not) for the consuming component to render this component conditionally?

Reply:
for some reason, the `isOpen` prop didn't actually seem to do anything, therefore this conditional render. But I'll come back to look at this